### PR TITLE
checkSubstitutionCoverage: check for constructor-like configuration

### LINKED ISSUE #2010: Unable to narrow on results of uninterpretted functions
The following definition crashes with (ErrorRewritesInstantiation)

```
module TEST
    imports BOOL
    imports INT

    syntax MaybeInt ::= "Some" Int
                      | "None"
    syntax MaybeInt ::= f() [function, functional, no-evaluators]

    configuration <k> f() </k>

    rule <k> None => true ... </k>
    rule <k> Some I => I ... </k>
endmodule
```

We expect it to narrow to two cases:

* one with `f() ==K None` and the K cell contains `true`
* one where `f() :=K Some I:Int` and the K cell contains `I`

### PR DESCRIPTION
**Pull request by andreiburdusa**
---
Fixes #2010 
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

**Comment by andreiburdusa** [comment]
I used a small `.k` file to test this and the [productivity](https://github.com/kframework/kore/issues/1988) was down to 87%

File: kore/src/Kore/Log/ErrorRewritesInstantiation.hs [hunk 1]
```diff
@@ -162,4 +170,11 @@ checkSubstitutionCoverage configuration solution
...
     isCoveringSubstitution = Set.null missingVariables
-    isSymbolic = Foldable.any isSomeConfigVariableName substitutionVariables
+    isSymbolic =
+        Foldable.any isSomeConfigVariableName substitutionVariables
+        || isNothing
+            ( getConstructorLike
+            . Attribute.constructorLikeAttribute
+            . extractAttributes
+            $ term configuration
+            )
```
Comment (ttuegel) [inline kore/src/Kore/Log/ErrorRewritesInstantiation.hs:180]: I think we only need to check that the configuration is not constructor-like:

```suggestion
    isSymbolic = (not . isConstructorLike) (term configuration)
```

File: test/imp/disjunction-in-simplification-spec.k.out.golden [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
-  <k>
-    x = choice ( a , b ); ~> .
-  </k>
-  <state>
-    a |-> A:Int
-    b |-> B:Int
-    x |-> _0
-  </state>
-</T>
+  #Not ( #Exists I . {
```
Comment (ttuegel) [inline test/imp/disjunction-in-simplification-spec.k.out.golden:1]: This test is not right. I think we should remove it.

**Comment by ttuegel** [comment]
One more thing: can you please add the example from the issue as an integration test?

**Pull request merged**